config/local-dependency-manager/src/package.ts

```diff
@@ -0,0 +1,226 @@
+    protected readonly rawLocalDependencies: {
+        [name: string]: string
+    };
+    protected readonly rawLocalDevDependencies: {
+        [name: string]: string
+    };
+
+    constructor(
+        readonly packagePath: string,
+        readonly fileWatcherProvider: FileWatcherProvider,
+        readonly includeDev: boolean
+    ) {
+        this.raw = require(path.resolve(packagePath, 'package.json')) || {};
+        this.rawLocalDependencies = !!this.raw.localDependencies ? this.raw.localDependencies : {};
+        this.rawLocalDevDependencies = this.includeDev && !!this.raw.localDevDependencies ? this.raw.localDevDependencies : {};
+    }
+
+    get name(): string {
+        if (this.raw.name) {
+            return this.raw.name;
+        }
+        return this.baseName;
+    }
+
+    get version(): string | undefined {
+        return this.raw.version;
+    }
+
+    get baseName(): string {
+        return path.basename(this.packagePath);
+    }
+
+    get archiveName(): string | undefined {
+        if (!this.version) {
+            return undefined;
+        }
+        return `${this.name}-${this.version}.tgz`;
+    }
+
+    get localPackages(): Package[] {
+        const packages: Package[] = [];
+        for (const dependency of this.localDependencies) {
+            const pck = this.getLocalPackage(dependency);
+            if (pck) {
+                packages.push(pck);
+            }
+        }
+        for (const dependency of this.localDevDependencies) {
+            const pck = this.getLocalPackage(dependency);
+            if (pck) {
+                packages.push(pck);
+            }
+        }
+        return packages;
+    }
+
+    getLocalPackage(dependency: string): Package | undefined {
+        const localPath = this.getLocalPath(dependency);
+        return localPath ? new Package(localPath, this.fileWatcherProvider, this.includeDev) : undefined;
+    }
+
+    getLocalPath(dependency: string | undefined): string | undefined {
+        if (!dependency) {
+            return undefined;
+        }
+        if (this.rawLocalDependencies[dependency]) {
+            return this.asLocalPath(this.rawLocalDependencies[dependency]);
+        }
+        return this.asLocalPath(this.rawLocalDevDependencies[dependency]);
+    }
+
+    protected asLocalPath(value: string | undefined): string | undefined {
+        return !!value ? this.resolvePath(value) : undefined;
+    }
+
+    get localDependencies(): string[] {
+        return Object.keys(this.rawLocalDependencies);
+    }
+
+    get localDevDependencies(): string[] {
+        return Object.keys(this.rawLocalDevDependencies);
+    }
+
+    get files(): string[] {
+        if (this.raw.files) {
+            return this.raw.files;
+        }
+        return ['lib', 'src'];
+    }
+
+    resolvePath(localPath: string): string {
+        return path.normalize(path.resolve(this.packagePath, localPath));
+    }
+
+    getNodeModulePath(dependency: undefined): undefined;
```

Comment:
I will remove there are no real clients for it, but it mostly to ensure that if you call with defined value then you never get `undefined`.